storage/src/main/java/org/apache/kafka/storage/internals/log/CleanShutdownFileHandler.java

```diff
@@ -85,17 +83,16 @@ void write(long brokerEpoch, int version) throws Exception {
         }
     }
 
-    public long read() {
-        long brokerEpoch = -1L;
+    @SuppressWarnings("unchecked")
+    public OptionalLong read() {
         try {
             String text = Utils.readFileAsString(cleanShutdownFile.toPath().toString());
-            Map<String, String> content = new ObjectMapper().readValue(text, HashMap.class);
-
-            brokerEpoch = Long.parseLong(content.getOrDefault(Fields.BROKER_EPOCH.toString(), "-1L"));
+            Content content = new ObjectMapper().readValue(text, Content.class);
```

Comment:
Hmm, by doing this, it seems that we can't upgrade from the current clean shutdown file nor could we downgrade if we add a new field in the future?

Reply:
You are right, downgrade is not flexible here. New fields will break the parser. Will revert the change.

Reply:
There is both @JsonIgnoreProperties(ignoreUnknown) and @JsonIgnore.

Reply:
Thanks for the tips!